drivers/LargeBlockSR.py

```diff
@@ -0,0 +1,203 @@
+    'description': 'SR plugin which emulates a 512 bytes disk on top of a 4KiB device then create a EXT SR',
+    'vendor': 'Vates',
+    'copyright': '(C) 2024 Vates',
+    'driver_version': '1.0',
+    'required_api_version': '1.0',
+    'capabilities': CAPABILITIES,
+    'configuration': CONFIGURATION
+}
+
+LARGEBLOCK_PREFIX = "XSLocalLargeBlock-"
+
+class LargeBlockSR(EXTSR.EXTSR):
+    """Emulating 512b drives for EXT storage repository"""
+
+    DRIVER_TYPE = "largeblock"
+    LOOP_SECTOR_SIZE = 512
+
+    @staticmethod
+    def handles(srtype):
+        return srtype == LargeBlockSR.DRIVER_TYPE
+
+    def load(self, sr_uuid):
+        super(LargeBlockSR, self).load(sr_uuid)
+        self.is_deleting = False
+        self.vgname = LARGEBLOCK_PREFIX + sr_uuid
+        self.remotepath = os.path.join("/dev", self.vgname, sr_uuid)
+
+    def attach(self, sr_uuid):
+        if not self.is_deleting:
+            self._create_emulated_device()
+            if not self._is_vg_connection_correct(): # Check if we need to redo the connection by parsing `vgs -o vg_name,devices self.vgname`
+                self._redo_vg_connection() # Call redo VG connection to connect it correctly to the loop device instead of the real 4KiB block device
+        super(LargeBlockSR, self).attach(sr_uuid)
+
+    def detach(self, sr_uuid):
+        super(LargeBlockSR, self).detach(sr_uuid)
+        if not self.is_deleting:
+            self._destroy_emulated_device()
+
+    @deviceCheck
+    def create(self, sr_uuid, size):
+        base_devices = self.dconf["device"].split(",")
+        for dev in base_devices:
+            logical_blocksize = util.pread2(["blockdev", "--getss", dev])
+            if int(logical_blocksize) == 512:
+                raise xs_errors.XenError("LargeBlockIncorrectBlocksize", opterr="The logical blocksize of the device {} isn't incompatible with other SR types".format(dev))
+
+        try:
+            self._create_emulated_device()
+            super(LargeBlockSR, self).create(sr_uuid, size)
+        finally:
+            self._destroy_emulated_device(base_devices)
+
+    def delete(self, sr_uuid):
+        base_devices = self.dconf["device"].split(",")
+
+        self.is_deleting = True
+        try:
+            self._create_emulated_device()
+            super(LargeBlockSR, self).delete(sr_uuid)
+        except SR.SROSError as e:
+            util.SMlog(e)
+        finally:
+            self._destroy_emulated_device(base_devices)
+            self.is_deleting = False
+
+    @deviceCheck
+    def probe(self):
+        # We override EXTSR.probe because it uses EXT_PREFIX in this call
+        return lvutil.srlist_toxml(
+            lvutil.scan_srlist(LARGEBLOCK_PREFIX, self.dconf['device']),
+            LARGEBLOCK_PREFIX
+        )
+
+    def _create_loopdev(self, dev, emulated_path):
+        cmd = ["losetup", "-f", "-v", "--show", "--sector-size", str(self.LOOP_SECTOR_SIZE), dev]
+        self.loopdev = util.pread2(cmd).rstrip()
```

Comment:
This member variable is never used outside this function. Is it useful to keep it?